File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        x = 0
        s = 0
        i = 0
        while n>0: 
            if n%10 >0:
                x+= (n%10)*(10**i)
                i+=1
                s+= n%10
            n//=10
        return x*s

        

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        int x = 0;
        int s = 0;
        int i = 0;
        while(n>0) {
            if(n%10 >0) {
                x+= (n%10)*Math.pow(10,i++);
                s+= n%10;
            }
            n/=10;
        }
        return (long)x*s;
    }
}